golang/networks/network_context.go

```diff
@@ -172,37 +161,50 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 	}
 	resp, err := networkCtx.client.StartService(ctx, startServiceArgs)
 	if err != nil {
-		return nil, nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+		return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
 	}
 	logrus.Tracef("Successfully started service with Kurtosis API")
 
-	logrus.Tracef("Creating service interface...")
-	service := containerCreationConfig.GetServiceCreatingFunc()(serviceContext)
-	logrus.Tracef("Successfully created service interface")
-
-	networkCtx.services[serviceId] = service
-
-	availabilityChecker := services.NewDefaultAvailabilityChecker(serviceId, service)
-
-	return service, resp.UsedPortsHostPortBindings, availabilityChecker, nil
+	return serviceContext, resp.UsedPortsHostPortBindings,  nil
 }
 
 // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-func (networkCtx *NetworkContext) GetService(serviceId services.ServiceID) (services.Service, error) {
-	networkCtx.mutex.Lock()
-	defer networkCtx.mutex.Unlock()
+func (networkCtx *NetworkContext) GetServiceContext(serviceId services.ServiceID, configFactory services.ContainerConfigFactory) (*services.ServiceContext, error) {
+	getServiceInfoArgs := &core_api_bindings.GetServiceInfoArgs{
+		ServiceId: string(serviceId),
+	}
+	serviceResponse, err := networkCtx.client.GetServiceInfo(context.Background(), getServiceInfoArgs)
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred when trying to get info from service '%v'",
+			serviceId)
+	}
 
-	service, found := networkCtx.services[serviceId]
-	if !found {
-		return nil, stacktrace.NewError("No service info found for ID '%v'", serviceId)
+	if serviceResponse.GetIpAddr() == "" {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred when trying to get the IP address from service '%v'",
+			serviceId)
 	}
-	return service, nil
+
+	containerCreationConfig, err := configFactory.GetCreationConfig(serviceResponse.GetIpAddr())
```

Comment:
Ooooo this is tough - I don't think we want to do this (trying to re-create the ContainerCreationConfig that was used to start the service), as it will be very tricky and error-prone. I'd much prefer to either a) remove ServiceContext entirely or b) store the necessary parts of ServiceContext inside the API container so that ServiceContext can be completely constructed from the GetServiceInfo response

Reply:
We will change this in the next PR

Reply:
I removed containerCreationConfig from here, it was replaced with GetSuiteExecutionVolMntDirpath()